core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -29,7 +29,26 @@ sealed abstract class Aggregate extends Data {
     val resolvedDirection = SpecifiedDirection.fromParent(parentDirection, specifiedDirection)
     val duplicates = getElements.groupBy(identity).collect { case (x, elts) if elts.size > 1 => x }
     if (!duplicates.isEmpty) {
-      throw new AliasedAggregateFieldException(s"Aggregate $this contains aliased fields $duplicates")
+      this match {
+        case b: Bundle =>
+          // show groups of names of fields with duplicate id's
+          // sorts are to make order of fields shown, deterministic and matching order of occurrence in bundle
+          // it's a bit convoluted but happens rarely and makes error easier to understand for user
+          val dupNames = duplicates.toSeq.sortBy(_._id).map { duplicate =>
+            b.elements
+              .collect { case x if x._2._id == duplicate._id => x }
+              .toSeq.sortBy(_._2._id)
+              .map(_._1).reverse
+              .mkString("(", ",", ")")
+          }.mkString(",")
+          throw new AliasedAggregateFieldException(
+            s"Bundle $this contains aliased fields named ${dupNames}"
+          )
+        case _ =>
+          throw new AliasedAggregateFieldException(
+            s"Aggregate $this contains aliased fields $duplicates ${duplicates.mkString(",")}"
```

Comment:
can we show hitting this in your test case below? I think the existing case only hits the above check?

Reply:
You could hit the bottom with a `Record`, but I think it would be better for the logic for `Record` to be the same as for `Bundle` up above. You can use value `className` in the String error to give more precise `Record` vs. concrete `Bundle` subclass in the error message.

If you fold `Record` handling into the above, this should be unreachable but still good to leave with a decent error message.

Reply:
I changed code to check for Record which will include Bundle.
I think the `case _ =>` should be considered future proofing against changes that inadvertently allow aliased fields or new types that might have them